### PR TITLE
Add the new inspec truncation options

### PR DESCRIPTION
**Pull request by alexpop**
This PR makes use of the latest inspec `reporter_backtrace_inclusion` and `reporter_message_truncation` [options](https://github.com/inspec/inspec/pull/4994) when triggering Compliance scans from Automate.

It also truncates the controls to maximum 50 (default) results per control before sending the report through ingestion. This is to prevent reports that exceed 4MB which are not allowed through grpc.
^ To tests this, you have to trigger a scan with a control that has a loop inside and generates lots of describe blocks.

### CODE REVIEW

**Comment by susanev** [comment]
should something be added to the automate docs about this?

**Comment by alexpop** [comment]
Spot on @susanev
I have another PR that will add a third cli option and was thinking to raise a doc request for all three.

**Comment by vjeffrey** [comment]
do you by any chance have a sample report/control with a bunch of results that ppl can use if they want to test?

File: components/compliance-service/inspec/cli.go [hunk 1]
```diff
@@ -299,6 +304,10 @@ func getInspecError(stdOut string, stdErr string, err error, target *TargetConfi
 		}
 	}
 
+	if err == nil {
```
Comment (vjeffrey) [inline components/compliance-service/inspec/cli.go:307]: don't we already cover this on line 321 with the return?

Comment (alexpop) [reply components/compliance-service/inspec/cli.go:307]: if err is nil, line 321 will not be reached as the service will crash on line 311.

Comment (vjeffrey) [reply components/compliance-service/inspec/cli.go:307]: ahh ya - 👍

**Comment by alexpop** [comment]
@vjeffrey here's a control that has 102 results (1 passed, 1 skipped, 100 failed):

```ruby
control 'mix control' do
  title 'Mix control to try to get different results'
  impact 1
  describe file("/etc/group") do
    it { should be_file }
  end
  describe apache_conf('path') do
    its('setting_name') { should eq 'value' }
  end
  (1..100).each do |i|
    describe file("/etc/group#{i}") do
      it { should be_file }
    end
  end
end
```

**Comment by vjeffrey** [comment]
thank you!

**Comment by alexpop** [comment]
Victoria, I opened another PR that has another service config option:
https://github.com/chef/automate/pull/3697

Once it merges I'll create a PR for all options introduced recently and then make sure documentation is also in sync.

**Pull request merged**